libraries/BLE/src/BLEDevice.h

```diff
@@ -0,0 +1,414 @@
+/*
+  BLE Device API
+  Copyright (c) 2016 Arduino LLC. All right reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
+#ifndef ARDUINO_BLE_DEVICE_H
+#define ARDUINO_BLE_DEVICE_H
+
+#include <Arduino.h>
+
+//#include "BLEService.h"
+//#include "BLECharacteristic.h"
+
+//class BLEDevice;
+
+enum BLEDeviceEvent {
+  BLEConnected = 0,         // BLE device connected
+  BLEDisconnected = 1,      // BLE device disconnected 
+  BLEConParamUpdate = 2,    // Update the connection parameter 
+                            //  Connection update request in central
+                            //  Connection parameter updated in peripheral
+  BLEDeviceLastEvent
+};
+
+typedef void (*BLEDeviceEventHandler)(BLEDevice device);
+
+class BLEDevice
+{
+  public:
+    /**
+     * @brief   The BLE device constructure
+     *
+     * @param   bleaddress  BLE device address
+     *
+     * @return  none
+     *
+     * @note  none
+     */
+    BLEDevice();
+    //BLEDevice(String bleaddress);
+    //BLEDevice(const char* bleaddress);
+    BLEDevice(const bt_addr_le_t* bleaddress);
+    BLEDevice(const BLEDevice* bleaddress);
+    
```

Comment:
BLEDevice(const BLEDevice* bleaddress); will be public. And the BLEDevice(const bt_addr_le_t* bleaddress); will be protected.